src/arrays/multidimensional_arrays/hard/q6/MainTest.java

```diff
@@ -0,0 +1,102 @@
+package arrays.multidimensional_arrays.hard.q6;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public static final int [][] a = { {4, 5, 6}, {66, 54, 21}, {609, 998, 543}, {1111, 2342, 4583}  };
+    public static final int [][] b = { {4433, 2316, 9879}, {662, 154, 721}, {69, 98, 54},{1, 4, 3}  };
+
+    public Clause[] testSentence() {
+        if(a.length != b.length || a[0].length != b[0].length) {
+            return new Clause[]{
+                    new StringLiteral("Can't add because arrays of different sizes!")
+            };
+        }
+        else{
+            Clause[] c = new Clause[3 * a.length];
+            int[][] ans = answerFor(a, b);
+            for(int i = 0; i < a.length; i ++){
+                String st = "";
+                for(int j = 0; j < a[0].length; j ++){
+                    st += ans[i][j] + " ";
+                }
+                c[3 * i] = new StringLiteral(st.trim());
+                c[3 * i + 1] = new Optional(new StringLiteral(" "));
+                c[3 * i + 2] = new NewLine();
+            }
+            return c;
+        }
+    }
+
+    public void runMain() {
+        OneAndOneMakeTwoDArrays.main(new String[0]);
+    }
+
+    public static int[][] answerFor(int[][] x, int[][] y){
+        if( x.length != y.length  || x[0].length != y[0].length )
+            return null;
+        int[][] ans = new int[x.length][x[0].length];
+        for(int i = 0; i < x.length; i ++){
+            for(int j = 0; j < x[0].length; j ++)
+                ans[i][j] = x[i][j] + y[i][j];
+        }
+        return ans;
+    }
+
+    static Stream<Arguments> inputProvider() {
```

Comment:
You can hardcode the answers to this instead of relying on a method

Reply:
I will still need the method for testSentence for testing the output of program, it looked more neat to do the same here is there any emphasis on hardcoding the solution whenever possible?

Reply:
It's more of a thing because we try not to rely on machine code to generate our answers when we can write them out by hand (keyboard) and thus be sure that errors within the code aren't perpetuated. It doesn't interrupt any generated cases in inputProvider, so hard coding in the solutions for hard coded test cases is preferred.